src/utils/thread_helper.cpp

```diff
@@ -147,7 +147,7 @@ void ThreadHelper::EnergyScan(uint32_t aScanDuration, EnergyScanHandler aHandler
 exit:
     if (error != OT_ERROR_NONE)
     {
-        if (aHandler)
+        if (mEnergyScanHandler)
```

Comment:
When `error != OT_ERROR_NONE`, `mEnergyScanHandler` is likely the callback set by previous `EnergyScan`. Why are we calling the callback of previous `EnergyScan` ?